File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> myStack;

        for (char c : s) {
            if (myStack.empty()) {
                myStack.push(c);
            } else {
                 char cOnTop = myStack.top();

            switch(c) {
               
                case ')':
                    if (cOnTop != '(') {
                        return false;
                    } else {
                        myStack.pop();
                    }
                    break;
                case ']':
                    if (cOnTop != '[') {
                        return false;
                    } else {
                        myStack.pop();
                    }
                    break;
                case '}':
                    if (cOnTop != '{') {
                        return false;
                    } else {
                        myStack.pop();
                    }
                    break;
                default:
                    myStack.push(c);
            }
            }
           
        }
        return myStack.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        for c in s:
            if c == "[" or c == "{" or c == "(":
                stack.append(c)
            elif c == "]":
                if not stack or stack[-1] != "[":
                    return False
                stack.pop()
            elif c == "}":
                if not stack or stack[-1] != "{":
                    return False
                stack.pop()
            elif c == ")":
                if not stack or stack[-1] != "(":
                    return False
                stack.pop()
            
        return len(stack) == 0